pkg/deploy/lattice/listener_manager.go

```diff
@@ -35,24 +37,37 @@ func NewListenerManager(cloud lattice_aws.Cloud, latticeDataStore *latticestore.
 	}
 }
 
-func (s *defaultListenerManager) Create(ctx context.Context, listener *latticemodel.Listener) (latticemodel.ListenerStatus, error) {
-	glog.V(6).Infof("Creating listener >>>> %v \n", listener)
+func (d *defaultListenerManager) Cloud() lattice_aws.Cloud {
+	return d.cloud
+}
 
-	serviceStatus, err := s.latticeDataStore.GetLatticeService(listener.Spec.Name, listener.Spec.Namespace)
+type ListenerNameProvider struct {
+	l *latticemodel.Listener
+}
 
-	if err != nil {
-		errmsg := fmt.Sprintf("Service %v not found during listener creation", listener.Spec)
-		glog.V(6).Infof("Error during create listner %s \n", errmsg)
-		return latticemodel.ListenerStatus{}, errors.New(errmsg)
-	}
+func (r *ListenerNameProvider) LatticeName() string {
+	return utils.LatticeServiceName(r.l.Spec.Name, r.l.Spec.Namespace)
+}
```

Comment:
Does this method use lattice listener name?

When create listener, we use that format?  i.e., `listener.Spec.Name` should follow this format? "routeName-routeNS-port-protocol"

https://github.com/aws/aws-application-networking-k8s/blob/138c1126540432673cd0628ed401c07bcdf96ea2/pkg/gateway/model_build_listener.go#L149

Reply:
The ```LatticeName()``` is intended to return the Lattice service name, based off the original route values for the listener. I can see here in this context ```LatticeName()``` is causing confusion. I'll update this to be ```LatticeServiceName()``` or similar instead.

Reply:
Oh, Seems your code is correct, I confused by the  second param`listenerResourceName` and fifth param ` t.route.Name()` here

https://github.com/aws/aws-application-networking-k8s/blob/138c1126540432673cd0628ed401c07bcdf96ea2/pkg/gateway/model_build_listener.go#L152

Reply:
Yeah these naming bits get confusing